docs/design/2023-03-15-ticdc-storage-sink.md

```diff
@@ -0,0 +1,404 @@
+and scenarios.
+
+## Detailed Design
+
+### Storage path structure
+
+This section describes the storage path structure of data change records, metadata, and DDL events. 
+Using the csv protocol as an example, files containing row change events should be organized as follows: 
+
+```
+s3://bucket/prefix1/prefix2                 <prefix>
+├── metadata                                
+└── schema1                                 <schema>
+    └── table1                              <table>
+        ├── 10000                           <table-version-separator>
+            ├── 13                          <partition-separator, optional>
+            │   ├── 2023-03-23              <date-separator, optional>
+            │   │   ├── CDC000001.csv
+            │   │   └── CDC.index
+            │   └── 2023-03-24              <date-separator, optional>
+            │       ├── CDC000001.csv
+            │       └── CDC.index
+            └── schema.json
+```
+
+#### Data change records
+
+Data change records are saved to the following path:
+
+```shell
+{scheme}://{prefix}/{schema}/{table}/{table-version-separator}/{partition-separator}/{date-separator}/CDC{num}.{extension}
+```
+
+- `scheme`: specifies the data transmission protocol, or the storage type, for example, <code>**s3**://xxxxx</code>.
+- `prefix`: specifies the user-defined parent directory, for example, <code>s3://**bucket/prefix1/prefix2**</code>.
+- `schema`: specifies the schema name, for example, <code>s3://bucket/prefix1/prefix2/**schema1**</code>.
+- `table`: specifies the table name, for example, <code>s3://bucket/prefix1/prefix2/schema1/**table1**</code>.
+- `table-version-separator`: specifies the separator that separates the path by the table version, for example, <code>s3://bucket/prefix1/prefix2/schema1/table1/**9999**</code>.
+- `partition-separator`: specifies the separator that separates the path by the table partition, for example, <code>s3://bucket/prefix1/prefix2/schema1/table1/9999/**20**</code>.
+- `date-separator`: classifies the files by the transaction commit date. Value options are:
+    - `none`: no `date-separator`. For example, all files with `test.table1` version being `9999` are saved to `s3://bucket/prefix1/prefix2/schema1/table1/9999`.
+    - `year`: the separator is the year of the transaction commit date, for example, <code>s3://bucket/prefix1/prefix2/schema1/table1/9999/**2022**</code>.
+    - `month`: the separator is the year and month of the transaction commit date, for example, <code>s3://bucket/prefix1/prefix2/schema1/table1/9999/**2022-01**</code>.
+    - `day`: the separator is the year, month, and day of the transaction commit date, for example, <code>s3://bucket/prefix1/prefix2/schema1/table1/9999/**2022-01-02**</code>.
+- `num`: saves the serial number of the file that records the data change, for example, <code>s3://bucket/prefix1/prefix2/schema1/table1/9999/2022-01-02/CDC**000005**.csv</code>.
```

Comment:
When does the serial number increase, is there any restriction to the size of single csv file?

Reply:
Data change files are saved to the downstream when either flush-interval or file-size meets the requirements. Ref: https://docs.pingcap.com/tidb/stable/ticdc-sink-to-cloud-storage#configure-nfs